### ISSUE TITLE
Additional hypervisors in running_in_a_vm?

### ISSUE THREAD

**Issue by clburlison** [Microsoft/macos-cookbook#61]
Based off these lines

https://github.com/Microsoft/macos-cookbook/blob/9b812e2e429f26aefb2753fbedfc8185ea2ed881/libraries/systemsetup.rb#L3-L5

it sounds like support for additional hypervisors would be a welcome PR?

Currently, the macos cookbook matching off of the `['hardware']['machine_model']` fact which will work on the default Parallels and VMware setup but not so much if the user overrides the model. Lots of admins override the model for testing purposes, mostly for Apple Device Enrollment Program tests. 

### Option 1
Instead of relying on the machine model to accurately represent the machine is a VM we could use the `['cpu']['flags']` output to look for `vmm` which is an accurate way to determine if the machine is a VM in VMWare Fusion, Parallels, VirtualBox or physical.

This is a more generic approach that matches the method name

### Option 2
We could use the `['ohai hardware']['boot_rom_version']` to accurately grab the VM type for VMWare Fusion & Virtualbox then shell out to system profiler to get the `SPEthernetDataType` key to validation on Parallels. See the IT-CPE PR for reference.

Thoughts? I personally prefer Option 1 but could see why you might want to know what hypervisor the machine is running.

References:
- [DEP testing in Parallels](https://jerbecause.wordpress.com/2018/02/09/creating-a-dep-vm-using-parallels-desktop/) 
- [DEP testing in VMware Fusion](https://www.rderewianko.com/how-to-create-a-vm-thatll-work-with-dep-on-vmware-fusion/)
- https://github.com/facebook/IT-CPE/pull/89/files
- https://github.com/munki/munki-facts/blob/master/facts/physical_or_virtual.py

**Comment by tas50** [comment]
Is this information not in the ohai Virtualization plugin?  You should be able to look for items in the node['virtualization']['systems'] mash that are of the type 'guest'. If parallels isn't properly detecting there please let me know over in the Ohai repo and I'll get that fixed up.

**Comment by jazaval** [comment]
We didn't know it existed - thanks! We'll aim to get this resolved in next week's release.

**Comment by americanhanko** [comment]
Thanks, @clburlison for the issue submission and @tas50 for that tip.

@jazaval This should do it:

```ruby
node['virtualization']['systems'].values.include? 'guest'
```

**Comment by tas50** [comment]
If something is missing just ping me on slack. I'm around. This cookbook is getting great and super important to us.

**Comment by clburlison** [comment]
I honestly didn't know the ohai virtualization code was designed to detect VM guests. Looking through the current ohai [virtualization code](https://github.com/chef/ohai/blob/3ffe07a3415a726a17b1bdc4626cabd32d986354/lib/ohai/plugins/darwin/virtualization.rb) Parallels is the only hypervisors that supports guest detection. So I opened https://github.com/chef/ohai/issues/1163

For what it is worth using the following block works right now with the currently shipping chef code:

```ruby
Chef.node['cpu']['flags'].include? 'vmm'
```

**Comment by americanhanko** [comment]
This is resolved via #73 - @clburlison if `node['virtualization']['systems']` returns an empty hash, we assume are not running on bare metal.
